packages/frontend/src/components/UserSettings/PasswordPanel/index.tsx

```diff
@@ -53,6 +53,22 @@ const PasswordPanel: FC = () => {
             },
         });
 
+    const setFormValuesFromData = useCallback(() => {
+        form.setValues({
+            currentPassword: '',
+            newPassword: '',
+        });
+        form.resetDirty({
+            currentPassword: '',
+            newPassword: '',
+        });
```

Comment:
hey @jogeshwar01 why do we need to set default value to the form and then resetDirty? can you explain how this works?

Reply:
this seems to me that it's serving two purposes, one is initializing the data and another one is resetting `dirty` state. I think this should be called `handleFormReset` and should be called from cancel button only. LMK!

Reply:
@IrakliJani The `form.setValues` is used to track the current values in the form while the `form.resetDirty` controls our current state that also helps us to display/hide the Cancel button on change

For instance, once a user clicks on `Save changes/Update` , we would want to set both our form values as well as the dirty state to the current changed state (to prevent Cancel button from showing after state has been updated)

So basically, we need this to be called in 2 cases - either user Saves the changes (useEffect) or cancels them.

Reply:
In case of Password Panel, as we redirect to /login when we update the password, we call it from the Cancel button only.
There are no dependencies for the `handleFormReset useCallback` and so it would never go inside the useEffect except  for the first time.